### DISCUSSION TITLE
Lesson 4 : Warning with install web3

### DISCUSSION THREAD

**Discussion by wilonweb** [smartcontractkit/full-blockchain-solidity-course-py#881] [Q&A] [answered]
Hello 
I did 
`pip install web 3`
and i got some warning 
`  WARNING: The script netaddr.exe is installed in 'C:\Users\wilfr\AppData\Roaming\Python\shttpclient, eth-account, web3Python39\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use 
--no-warn-script-location.
  WARNING: The script base58.exe is installed in 'C:\Users\wilfr\AppData\Roaming\Python\PPython39\Scripts' which is not on PATH.ython39\Scripts' which is not on PATH.                                                   --no-warn-script-location.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use ython39\Scripts' which is not on PATH.
--no-warn-script-location.                                                               --no-warn-script-location.
    Running setup.py install for bitarray ... done
    Running setup.py install for lru-dict ... done
  WARNING: The script jsonschema.exe is installed in 'C:\Users\wilfr\AppData\Roaming\Python\Python39\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use 
--no-warn-script-location.`

and when i put 
`form web3 import Web3` my title file deploy.py become yellow.

I have to change something or is okay ?

**Comment by cromewar** [comment, marked as answer]
Hello @NinjaTurtles-cloud 
I think is a typo, but here pip install web 3 you have a space between web and 3, so be sure you install it like this:
`pip install web3`
That warnings are related to env variables, they are not actual errors more like recommendations.
On VsCode on the bottom left corner make sure to select the correct python interpreter (the one you installed web3).

Finally some recommendations:

1. use virtualenv for your python projects.
2. use python version 3.8 (is the most stable for web3 dev).

I hope this info might help.

**Reply by wilonweb** [reply]
Hello @cromewar 
Thanks for your answer. 

Sorry i had written well
`pip install web3` on VsCode
On the bottom left corner i see Python 3.9.6 and if i click on it i can switch the interpreter path like [the screenshot](https://ibb.co/cLWX2rq).

I have to switch on python version 3.8 to follow the tutorial ? 

I have to inquire to know how use a virtual env. 
Do you think I can still continue the tutorial without installing a virtual env for now ?

**Reply by cromewar** [reply]
virtualenv envs i just a way to have isolates instances of python in order to avoid dependency conflicts, but is not an obligation.
The version must be the same one as you have on the console, if you type python --version you'll find which version is the correct one.
